samza-sql/src/main/java/org/apache/samza/sql/translator/QueryTranslator.java

```diff
@@ -151,7 +155,11 @@ public RelNode visit(LogicalAggregate aggregate) {
 
     Optional<TableDescriptor> tableDescriptor = sinkConfig.getTableDescriptor();
     if (!tableDescriptor.isPresent()) {
-      outputStream.sendTo(streamGraph.getOutputStream(sinkConfig.getStreamName()));
+      KVSerde<Object, Object> noOpKVSerde = KVSerde.of(new NoOpSerde<>(), new NoOpSerde<>());
+      String systemName = sinkConfig.getSystemName();
+      InternalSystemDescriptor sd = context.getSystemDescriptors().computeIfAbsent(systemName, InternalSystemDescriptor::new);
```

Comment:
Quick question: I think SqlIOConfig has included all configuration for systems in sinkConfig object. What's the relationship of this InternalSystemDescriptor and the sinkConfig object? The Javadoc for InternalSystemDescriptor says "for systems that the factory class is not know in advance". So, I assume this InternalSystemDescriptor is just a dummy reference to the actual configuration for systems in sinkConfig? Hence, in runtime, this dummy InternalSystemDescriptor does not contain any actual configuration and will be merged w/ the configuration already in sqlConfig. Can we have a different name for this SystemDescriptor, like DelegatingSystemDescriptor? InternalSystemDescriptor is too vague.

Reply:
Yeah, that's a recent change. Full job configs aren't currently wired up everywhere. E.g., ConfigBasedIOResolver doesn't use it. Even if it were, this class would be useful for providing a simple "pass-through" system descriptor for testing and code-generation use cases. Your explanation of how this works is correct. I'll rename it to DelegatingSystemDescriptor.